### PR TITLE
When checking signatures, fail gracefully in the face of more db errors

### LINKED ISSUE #1971: Modify signature check to gracefully fail on db schema failure
If the signature check attempts to access the database and fails, potentially because the db is unvailable or because some schema is not available, the check should warn and pass, and a separate db check will fail.  This prevents a circular dependency between migrations running and the checks depending on them.

### PR DESCRIPTION
**Pull request by mythmon**
Fixes #1971

### CODE REVIEW

File: normandy/recipes/checks.py [hunk 1]
```diff
@@ -50,8 +51,11 @@ def recipe_signatures_are_correct(app_configs, **kwargs):
         signed_recipes = list(Recipe.objects.exclude(signature=None).select_related("signature"))
     except (ProgrammingError, OperationalError, ImproperlyConfigured) as e:
         errors.append(Info(f"Could not retrieve recipes: {e}", id=INFO_COULD_NOT_RETRIEVE_RECIPES))
-    else:
+        return errors
+
+    try:
         for recipe in signed_recipes:
+            print("@@@", recipe.canonical_json)
```
Comment (rehandalal) [inline normandy/recipes/checks.py]: did you intend to have this in here?

**Comment by jaredkerim** [comment]
I was able to recreate the problem locally and was able to apply migrations with this patch, so lgtm 👍

File: normandy/recipes/checks.py (hunk 1 above)
Comment (jaredkerim) [reply normandy/recipes/checks.py]: I'll wait for @mythmon to respond to this before we merge.

**Comment by jaredkerim** [comment]
bors try

File: normandy/recipes/checks.py (hunk 1 above)
Comment (mythmon) [reply normandy/recipes/checks.py]: Oops. Fixed.

**Comment by mythmon** [comment]
bors r= @jaredkerim

**Pull request merged**